src/crimson/osd/osd_operations/client_request.cc

```diff
@@ -317,21 +323,33 @@ ClientRequest::process_op(
       "Skipping recover_missings on non primary pg for soid {}",
       *pg, m->get_hobj());
   } else {
-    co_await do_recover_missing(pg, m->get_hobj().get_head(), m->get_reqid());
+    auto unfound = co_await do_recover_missing(
+      pg, m->get_hobj().get_head(), m->get_reqid());
+    if (unfound) {
+      DEBUGDPP("{} unfound, hang it for now", *pg, m->get_hobj().get_head());
+      co_return;
+    }
+
     std::set<snapid_t> snaps = snaps_need_to_recover();
     if (!snaps.empty()) {
       // call with_obc() in order, but wait concurrently for loading.
       ihref.enter_stage_sync(
           client_pp(*pg).recover_missing_lock_obc, *this);
       auto with_obc = pg->obc_loader.with_obc<RWState::RWREAD>(
         m->get_hobj().get_head(),
-        [&snaps, &ihref, pg, this](auto head, auto) {
-        return recover_missing_snaps(pg, ihref, head, snaps);
+        [&snaps, &ihref, pg, this, &unfound](auto head, auto) {
+        return recover_missing_snaps(pg, ihref, head, snaps
+        ).then_interruptible([&unfound](bool r) {
+          unfound = r;
+        });
       }).handle_error_interruptible(
         crimson::ct_error::assert_all("unexpected error")
       );
       // see https://gcc.gnu.org/bugzilla/show_bug.cgi?id=98401
       co_await std::move(with_obc);
+      if (unfound) {
+        co_return;
```

Comment:
nit: debug logs

Reply:
Are we simply dropping the request or am I missing something?

Reply:
You can't drop the request -- it needs to wait for the object to no longer be unfound.

Reply:
will correct this.